docs/reference/contributing/target/rtos.md

```diff
@@ -1,75 +1,31 @@
-`CMSIS_5/CMSIS/RTOS2/RTX/Include/` | `mbed-os/rtos/TARGET_CORTEX/rtx5` | RTX definitions |
-`CMSIS_5/CMSIS/RTOS2/RTX/Source/rtx_*` | `mbed-os/rtos/TARGET_CORTEX/rtx5` | RTX sources |
-`CMSIS_5/CMSIS/RTOS2/RTX/Source/svc_user.c` | `mbed-os/rtos/rtx2/TARGET_CORTEX_M/` | RTX SVC user table |
-`CMSIS_5/CMSIS/RTOS2/RTX/Source/{ARM,GCC,IAR}/` | `mbed-os/rtos/TARGET_CORTEX/rtx5/TARGET_{M0,M0P,M3,RTOS_M4_M7,M23,M33}/TOOLCHAIN_{ARM,GCC,IAR}` | Toolchain and core specific exception handlers |
+Due to differences in the Mbed OS and CMSIS directory structure, you can't import the original code directly. You should use the `importer.py` and configuration file `cmsis_importer.json` to import upstream CMSIS code.
 
 #### Modification
 
-Due to different use cases between Mbed OS and CMSIS, we had to make some modifications to the source code. We've tried to upstream our changes to the CMSIS repository, but in cases where they aren't compatible with CMSIS requirements, we are forced to maintain a small set of changes.
-
-##### CMSIS
-
-Filename | Description |
----------|-------------|
-`cmsis_compiler.h` | Added IAR missing __ALIGNED attribute for earlier (less than 7.8.4) versions |
-`cmain.S` | custom IAR non-RTOS boot sequence for Mbed |
-
-##### RTX
-
-Filename | Description |
----------|-------------|
-`cmsis_os2.h` | Doxygen added; added per-thread uVisor context |
-`cmsis_os1.h` | Change `osThreadDef` to accept 3 parameters rather than 4 and be not static as expected by Mbed OS |
-`core_cm.h` | Doxygen added; included headers changed to match Mbed OS core selection; deferred priority setting of SVCall to uVisor when uVisor is enabled |
-`RTX_Config.h` | Doxygen added; Mbed OS RTX config included |
-`rtx_evr.c` | CMSIS component definition include removed |
-`rtx_evr.h` | Doxygen added |
-`rtx_thread.c` | Added per-thread uVisor context; notify uVisor of OS events  |
-`rtx_kernel.c` | Added per-thread uVisor context; notify uVisor of OS events |
-`rtx_lib.h` | Doxygen added; added per-thread uVisor context |
-`rtx_os.h` | Doxygen added; added per-thread uVisor context |
-`irq_cm4.s` | For all toolchains: added case for Cortex M4 cores without VFP |
-`svc_user.c` | Removed as its template file and should not be in our code base |
-`rt_OsEventObserver.{c,h}` | Added an interface for uVisor to be notified about certain events from privileged code |
-`irq_armv8mbl.S` | IAR toolchain: added file for Cortex M23 core |
-`irq_armv8mml.S` | IAR toolchain: added file for Cortex M33 core |
-
-<span class="notes">**Note:** For all toolchains, Mbed OS uses `irq_cm0.s` for both M0 and M0P cores.</span>
+Due to different use cases between Mbed OS and CMSIS, we modified the source code. We upstream our changes to the CMSIS repository, but in cases when they aren't compatible with CMSIS requirements, we maintain a small set of changes. We maintain changes as separate commits in `mbed-os`, and SHAs are in the `commit_sha` section of the `cmsis_importer.json` file.
```

Comment:
> Due to different use cases between Mbed OS and CMSIS, we modified the source code

that's a bit vogue, which source code?

> We upstream our changes to the CMSIS repository, but in cases when they aren't compatible with CMSIS requirements, we maintain a small set of changes. We maintain changes as separate commits in `mbed-os`, and SHAs are in the `commit_sha` section of the `cmsis_importer.json` file.

That is a bit confusing. @AnotherButler what do you think?

Reply:
@bulislaw What part do you think is confusing?

Reply:
Maybe confusing isn't the best word for that, but we repeat ourselves and it makes it feel like we talk about something else (?)  `we maintain a small set of changes. We maintain changes` if I'm saying gibberish please ignore :)